### ISSUE TITLE
Add possibility to abort passes in bomber-ball

### ISSUE THREAD

**Issue by TheAssassin** [blue-nebula/base#232]
By holding the "pass key" (typically `F`), one can aim for a mate, and when the key is released, the bomb will follow the corresponding mate (though, slower than without aiming). This can be very useful in-game. However, once the key is held, the pass cannot be aborted any more. This can be a serious issue, for instance, when the desired mate disappears or dies. One needs to keep holding the key and hope for a better chance.

We should introduce a possibility for passes to be aborted. I have no good idea yet how to implement this, but most likely we need to add another keybind for this. We should preserve the "simple" pass.

**Comment by Lagetta** [comment]
My idea is not entirely create new bind key, but using the existing ones.
For example when you hold a button lets say "f", you get to see those circles where the ball will be passed, and click button like the same "shoot" button.
However with that, you will lose the ability to just throw the ball.

I am thinking that button "z" (default drop weapon) could serve as throw button. But then it comes the flags, I would keep that same "f" for dropping the flag.
Also, dropping weapon button is never used button at all. In classic mode, people just simply choose the weapon that they want to remove and replace it. it could be even removed entirely.

**Comment by TheAssassin** [comment]
Not sure whether that would work so well. I don't know how you could reach a button like Z while holding the F key already. I think you'd have to use the little finger, but then again, it's used for parkour a lot.

Perhaps we should just abort a pass when no target is in sight. That, however, adds a risk of not getting rid of the bomb in time.

**Comment by robalni** [comment]
A few ideas:

* We could invert it, so you pass if you press F and throw if you hold F.
* We could abort the pass if you switch weapons (just like you abort grenades) or if you shoot a weapon.
* We could abort the pass if there is no teammate close to the center of the screen, either because they disappeared or because you look away.
* We could have different keys for throwing and passing, like press F to pass and press G or C to throw.

**Comment by TheAssassin** [comment]
> We could invert it, so you pass if you press F and throw if you hold F.

I don't think that this will work. You can't aim that way. For instance, I try to aim for a real mate (ideally one I know knows the game mode), not a bot.

> We could abort the pass if there is no teammate close to the center of the screen, either because they disappeared or because you look away.

That's something I thought about, too. I think this must be combined with a kind of smaller (e.g., a circle with a diameter of 2/3 of the screen height) "aiming area" around the crosshair to reduce the risk of "accidental" passes.

> We could have different keys for throwing and passing, like press F to pass and press G or C to throw.

I think you'd still need an "abort" functionality while passing, though. It might help newbies to learn passing, though.
